src/mono/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.Mono.cs

```diff
@@ -32,8 +32,12 @@ public static void InitializeArray(Array array, RuntimeFieldHandle fldHandle)
         [Obsolete("OffsetToStringData has been deprecated. Use string.GetPinnableReference() instead.")]
         public static int OffsetToStringData
         {
-            [Intrinsic]
-            get => OffsetToStringData;
+            get =>
+#if TARGET_64BIT
+                20;
```

Comment:
Can we define an internal constant in String.Mono.cs instead and then refer to it by name here. That way the value is closer to the rest of the string implementation.